sklearn/tree/_tree.pyx

```diff
@@ -572,17 +587,25 @@ cdef class Tree:
         def __get__(self):
             return self._get_node_ndarray()['weighted_n_node_samples'][:self.node_count]
 
+    property missing_direction:
+        def __get__(self):
+            return self._get_node_ndarray()['missing_direction'][:self.node_count]
+
     property value:
         def __get__(self):
             return self._get_value_ndarray()[:self.node_count]
 
     def __cinit__(self, int n_features, np.ndarray[SIZE_t, ndim=1] n_classes,
-                  int n_outputs):
+                  int n_outputs, bint allow_missing, object missing_values,
+                  object random_state):
         """Constructor."""
         # Input/Output layout
         self.n_features = n_features
         self.n_outputs = n_outputs
         self.n_classes = NULL
+        self.allow_missing = allow_missing
+        self.missing_values = missing_values
+        self.random_state = random_state
```

Comment:
Again, I dont see why we need this. A tree structure is fixed and cannot evolve. If at all, the random state object should be stored as part of the Python `DecisionTree*` estimator.